docs/helm-chart/quick-start.rst

```diff
@@ -65,8 +65,17 @@ Run ``kubectl port-forward svc/airflow-webserver 8080:8080 -n airflow``
 to port-forward the Airflow UI to http://localhost:8080/ to confirm
 Airflow is working.
 
-Build a Docker image from your DAGs
------------------------------------
+Extending Airflow Image
```

Comment:
I have mixed feelings about this. This is a quick start guide for Helm Chart, and I think we should be focusing on that. For the same reason, we do not describe image building in the [Docker-compose guide](http://airflow.apache.org/docs/apache-airflow/stable/start/docker.html). Now over 60% of the content on this guide is not about Helm Chart but about the image. 

Helm Chart for Gitlab has a separate guide [for the custom image/images](https://docs.gitlab.com/charts/advanced/custom-images/index.html). What do you think about it?  They also have [a quick start guide](https://docs.gitlab.com/charts/quickstart/index.html).

Helm Chart for JupytterHub also have comprehensive documentation with [guide for custom image](https://zero-to-jupyterhub.readthedocs.io/en/latest/jupyterhub/customizing/user-environment.html#choose-and-use-an-existing-docker-image)

Reply:
I think this falls into a "quick start"  actually. Originally there was only "adding DAGs to your image" here. but I think it will be as common to add PyPI/Apt dependencies during the "quick start". If you add your DAGs, very likely you want to add dependency. This was the top ask from Helm chart users - in slack in elsewhere ("How can I add new dependency").

I think many users do not realize how easy it is for your local environment to build and deploy your image (and that is needed sooner-or-later most of the time anyway for Airflow). I think this was the main reason why @MarkusTeufelberger and others used the "additional Packages" option in the original helm chart  - they did not realized that they can easily add  PyPI/APT packages via custom image. I think this is the original fallacy of the "dynamic" installation method. You are tempted to do everything by helm chart configuration. 

Unfortunately (or fortunately depends how you look at it) Kubernetes + Helm  + Docker(Container) images  are all a  leaky-abstraction. You should understand (at least to some extent) all of it and be able to modify all of it when you want to deploy application via the helm chart.  So the docs has to be a little about all of it. By adding it where they will be looking for a "quick start" it's also a bit of educating the users that they can and should do it.

Also The "quick start" here is not a "generic" quick start. This is "Quick Start with kind" - very specific case, where you use `kind` as test-and-try platform (you should not  use `kind` in production - it is intended for test and development). Building and loading images in kind is just natural part of the flow with kind. And for the "quick start", you should find all the "quick need" answers on that single page rather than somewhere else (even if you could have a link). So I think adding the most common "quick-start" scenarios makes perfect sense, even if it involves building the image and not configuring helm chart. I thought a bit about this and I am 100% sure this is best place for it.

For production deployment - yes, there is no need to describe it there, simply reference to "Building the image" is enough as for production deployment you have more experience and time to read it.

Also GitLab is a bit different story. GitLab works standalone out-of-the-box for vast majority of cases. There are very little reasons you would like to build GitLab image. On the other hand - I cannot imagine a "serious" deployment of Airflow where you would use the "reference" image. You need to learn to build your own Airflow image - the sooner, the better.